index.js

```diff
@@ -62,7 +63,13 @@ function terser(userOptions = {}) {
 
       result.then(handler, handler);
 
-      return result;
+      return result.then(result => {
+        if (result.nameCache) {
+          merge(userOptions.nameCache, result.nameCache)
```

Comment:
`Object.assign` would actually be preferable here, as it will correctly remove nested keys that are no longer present in the updated `nameCache` object:

```suggestion
          Object.assign(userOptions.nameCache, result.nameCache)
```

Reply:
I would have but it's running on multiple threads. In that case, it needs to merge fully

Reply:
Hmm - that's a problem in general then, since it means properties will never be delete from the map.

Reply:
Internally the namecache is just updated, deletion isn't implemented by terser